### PR TITLE
Peer review 1 - ready to be reviewed

### PR DESCRIPTION
**Pull request by S-McD**

### CODE REVIEW

File: program/lib/password_validator.rb [hunk 1]
```diff
@@ -3,4 +3,9 @@
 # - include at least one of these characters: !, @, $, %, &
 
 def valid?(password)
+    if (password.length >=8) && (password =~ /[!@$%&]/)
```
Comment (d-r-holden) [inline program/lib/password_validator.rb:6]: I like the use of the use of the regular expression. It is something I wasn't aware of, and makes your code look very clear and efficient.

**Comment by d-r-holden** [comment]
Hi - After reviewing your code, I can't suggest and improvements which could be made. It appears to be efficiently written and well structured.

File: program/lib/password_validator.rb (hunk 1 above)
Comment (S-McD) [reply program/lib/password_validator.rb:6]: Thank you very much, I appreciate the feedback

**Comment by S-McD** [comment]
Peer review completed and branch merged back on to the main.
Saving the branch for tutor review and will then delete.

File: program/lib/password_validator.rb (hunk 1 above)
Comment (oceansize) [inline program/lib/password_validator.rb:6]: Hi Simone — nice job! I'm wondering if there's a way to make this read more clearly though. For example, when I see a regular expression, I realise I'm going to have to do some detective work to figure out what this particular one will do.

You can make life easier on your colleagues (and your future self) by abstracting a regex into a variable. This means your code might look something like this:

```ruby
def valid?(password)
  special_characters_to_match = /[!@$%&]/
  if (password.length >=8) && (password =~ special_characters_to_match)
        true
    else
        false
    end    
end
```

**Pull request merged**